certora/specs/AuctionManageSetup.spec

```diff
@@ -0,0 +1,59 @@
+methods {
+    function _.upgradeToAndCall(address,bytes) external => NONDET;
+}
+
+/// Is persistent because it mirrors the PoolManager storage.
+persistent ghost mapping(uint256 => uint256) bids_amount {
+    init_state axiom forall uint256 bid_id . bids_amount[bid_id] == 0;
+}
+
+ghost mathint sum_of_bids {
+    init_state axiom sum_of_bids == 0;
+}
+
+ghost mathint sum_of_active_bids {
+    init_state axiom sum_of_active_bids == 0;
+}
+
+ghost mathint sum_of_all_bids_amounts {
+    init_state axiom sum_of_all_bids_amounts == 0;
+}
+
+ghost mathint sum_of_all_active_bids_amounts {
+    init_state axiom sum_of_all_active_bids_amounts == 0;
+}
+
+ghost uint256 latest_bid_id;
+
+ghost mapping(uint256 => bool) bids_is_active {
+    init_state axiom forall uint256 bid_id . bids_is_active[bid_id] == false;
+}
+
+ghost mapping(address => uint256) user_last_bid_id_index {
+    init_state axiom forall address user . user_last_bid_id_index[user] == 0;
+}
+
+hook Sstore bids[KEY uint256 bid_id].amount uint256 new_amount (uint256 old_amount) {
+    bids_amount[bid_id] = new_amount;
+    sum_of_bids = sum_of_bids + 1;
```

Comment:
Is it possible to reset the bid? In that case when new_amount = 0 the sum_of_bids should reduce by 1.